Ginger/GingerCore/ALM/Qtest/ExportToQtest.cs

```diff
@@ -0,0 +1,594 @@
+                            || publishToALMConfig.FilterStatus == FilterByStatus.All)
+                            {
+                                TSTest tsTest = null;
+                                //go by TC ID = TC Instance ID
+                                tsTest = qcTSTests.Where(x => x.TestId == activGroup.ExternalID && x.ID == activGroup.ExternalID2).FirstOrDefault();
+                                if (tsTest == null)
+                                {
+                                    //go by Linked TC ID + TC Instance ID
+                                    tsTest = qcTSTests.Where(x => ImportFromQC.GetTSTestLinkedID(x) == activGroup.ExternalID && x.ID == activGroup.ExternalID2).FirstOrDefault();
+                                }
+                                if (tsTest == null)
+                                {
+                                    //go by TC ID 
+                                    tsTest = qcTSTests.Where(x => x.TestId == activGroup.ExternalID).FirstOrDefault();
+                                }
+                                if (tsTest != null)
+                                {
+                                    //get activities in group
+                                    List<Activity> activities = (bizFlow.Activities.Where(x => x.ActivitiesGroupID == activGroup.Name)).Select(a => a).ToList();
+                                    string TestCaseName = PathHelper.CleanInValidPathChars(tsTest.TestName); 
+                                    if ((publishToALMConfig.VariableForTCRunName == null) || (publishToALMConfig.VariableForTCRunName == string.Empty))
+                                    {
+                                        String timeStamp = DateTime.Now.ToString("dd-MMM-yyyy HH:mm:ss");
+                                        publishToALMConfig.VariableForTCRunName = "GingerRun_" + timeStamp;
+                                    }
+
+                                    RunFactory runFactory = (RunFactory)tsTest.RunFactory;
+                                    Run run = (Run)runFactory.AddItem(publishToALMConfig.VariableForTCRunNameCalculated);
+
+                                    // Attach ActivityGroup Report if needed
+                                    if (publishToALMConfig.ToAttachActivitiesGroupReport)
+                                    {
+                                        if ((activGroup.TempReportFolder != null) && (activGroup.TempReportFolder != string.Empty) &&
+                                            (System.IO.Directory.Exists(activGroup.TempReportFolder)))
+                                        {
+                                            //Creating the Zip file - start
+                                            string targetZipPath = System.IO.Directory.GetParent(activGroup.TempReportFolder).ToString();
+                                            string zipFileName = targetZipPath + "\\" + TestCaseName.ToString() + "_GingerHTMLReport.zip";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [There's no need to call 'ToString()' on a string.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)